libraries/rush-lib/src/schemas/build-cache.schema.json

```diff
@@ -148,7 +148,7 @@
           "properties": {
             "cacheProvider": {
               "type": "string",
-              "pattern": "^(?:(?!azure-blob-storage|amazon-s3|http).)*$"
+              "pattern": "^(?:(?!local-only|azure-blob-storage|amazon-s3|http).)*$"
```

Comment:
This entry is a catch-all entry that matches any value *except* "azure-blob-storage", "amazon-s3", or "http". Or at least, that's what it is supposed to be; in practice it doesn't allow those values to be substrings either.

Reply:
Yeah, sorry I just noticed that as well (the tests failed so I guess the blame is on me for rushing it a bit haha!)

Fix included in latest commit.

Reply:
I'm still not clear on what this fixes; the existing catch-all isn't an error case, so should allow "local-only" as a valid value.

Reply:
@dmichon-msft it fixes code completion in editors like VSCode when editing build-cache.json. Currently VSCode doesn't display "local-only" as valid option so it might be misleading/confusing for users.

Screenshot below:

![image](https://github.com/user-attachments/assets/5934b74d-224a-4043-b857-1257aee4656b)

(sorry for not being clear about this)

Reply:
![image](https://github.com/user-attachments/assets/d7027e1b-3c6e-4978-9d14-a97ee1a301d9)

(when using schema from this PR)